boards/pine64-pinephoneA64/default.nix

```
@@ -0,0 +1,56 @@
+{ config, lib, pkgs, ... }:
+
+let
+  pw = id: sha256: pkgs.fetchpatch {
+    inherit sha256;
+    name = "${id}.patch";
+    url = "https://patchwork.ozlabs.org/patch/${id}/raw/";
+  };
+in
+{
+  device = {
+    manufacturer = "PINE64";
+    name = "Pinephone (A64)";
+    identifier = "pine64-pinephoneA64";
+  };
+
+  hardware = {
+    soc = "allwinner-a64";
+  };
+
+  Tow-Boot = {
+    defconfig = "pinephone_defconfig";
+    phone-ux = {
+      enable = true;
+      blind = true;
+      wip = {
+        led_R = "led-2";
+        led_G = "led-1";
+        led_B = "led-0";
+        mmcSD   = "0";
+        mmcEMMC = "1";
+      };
+    };
+    config = [
+      (helpers: with helpers; {
+        BUTTON_GPIO = yes;
+        BUTTON_SUN4I_LRADC = yes;
+        LED_GPIO = yes;
+        VIBRATOR_GPIO = yes;
+      })
+      (helpers: with helpers; {
+        USB_MUSB_GADGET = yes;
+        USB_GADGET_MANUFACTURER = freeform ''"Pine64"'';
+      })
```

```suggestion
      })
      (helpers: with helpers; {
        SUPPORT_EMMC_BOOT = yes;
      })
```

This option would allow u-boot to boot from eMMC boot partitions on the A64.

---

Also, upstream u-boot has support for startup LED (since 2022.01 IIRC) in case you hadn't noticed, so an additional patch isn't needed for that (vibrator isn't supported upstream, however).

(I have to admit I didn't look up what u-boot version this PR was based on though, so this comment might or might not be relevant)

---

The patch implements LED startup like other RK3399 boards implement it.

I seem to remember looking at implementing it "correctly" through their framework, and that it could be used during SPL since it brings in too much code to fit.

---

> This option would allow u-boot to boot from eMMC boot partitions on the A64.

Stay tuned for the A64, work is in progress for support for eMMC boot partitions.

---

Oh, forgot to come back to this discussion and say it: it's done. It works. Tested against Allwinner and Amlogic targets.